### PR TITLE
Manage net resources (apply_net_config)

### PR DESCRIPTION
**Pull request by sjberman**
Adding support to properly manage net resources (currently ARP and FDB tunnels) on the BIG-IP device. This follows a similar pattern to the current apply_ltm_config and deploy steps. FDB tunnels are treated slightly different, as users may have pre-configured these and expect CCCL to update them. These types of resources will be updated only (no creation or deletion). CCCL also supports creation and deletion of new or existing tunnels if specified.

Also fixed a few bugs involving configuration of tunnels/records that resulted in unnecessary updates.

### CODE REVIEW

File: f5_cccl/bigip.py [hunk 1]
```diff
@@ -382,8 +382,7 @@ def _refresh_net(self):
         # Retrieve the list of tunnels
         LOGGER.debug(
             "Retrieving fdb tunnels from BIG-IP /%s...", self._partition)
-        tunnels = self._bigip.tm.net.fdb.tunnels.get_collection(
-            requests_params={"params": query})
+        tunnels = self._bigip.tm.net.fdb.tunnels.get_collection()
```
Comment (edarzins) [inline f5_cccl/bigip.py]: You removed the query (which was filtering on self._partition), but then you filter the results in L396-401 based on self._partition. I don't understand why you changed that -- should have worked the way it was.

Comment (sjberman) [reply f5_cccl/bigip.py]: I believe there is currently a bug in the sdk (or maybe I'm not understanding the functionality) for this specific case of "get_collection". FDB tunnels are tagged by the sdk with an icontrol_version, which processes the requests_params differently. It expects 'params' to be an object, not a string, but when I attempted to configure it as an object containing the filtered partition, I didn't have any success (the sdk didn't recognize the field). Have you experienced anything like this?

Here is the example traceback:
```
----> 1 mgmt.tm.net.fdb.tunnels.get_collection(requests_params={"params": "$filter=partition+eq+Common"})

/home/ubuntu/.virtualenvs/systest/lib/python2.7/site-packages/f5/bigip/resource.pyc in get_collection(self, **kwargs)
    777         """
    778         list_of_contents = []
--> 779         self.refresh(**kwargs)
    780         if 'items' in self.__dict__:
    781             for item in self.items:

/home/ubuntu/.virtualenvs/systest/lib/python2.7/site-packages/f5/bigip/resource.pyc in refresh(self, **kwargs)
    647         >>> print(resource_obj.raw)
    648         """
--> 649         self._refresh(**kwargs)
    650
    651     def load(self, **kwargs):

/home/ubuntu/.virtualenvs/systest/lib/python2.7/site-packages/f5/bigip/resource.pyc in _refresh(self, **kwargs)
    619     def _refresh(self, **kwargs):
    620         """wrapped by `refresh` override that in a subclass to customize"""
--> 621         requests_params = self._handle_requests_params(kwargs)
    622         refresh_session = self._meta_data['bigip']._meta_data['icr_session']
    623         if self._meta_data['uri'].endswith('/stats/'):

/home/ubuntu/.virtualenvs/systest/lib/python2.7/site-packages/f5/bigip/resource.pyc in _handle_requests_params(self, kwargs)
    334         if self._meta_data['icontrol_version']:
    335             params = requests_params.pop('params', {})
--> 336             params.update({'ver': self._meta_data['icontrol_version']})
    337             requests_params.update({'params': params})
    338         return requests_params

AttributeError: 'str' object has no attribute 'update'
```

Comment (edarzins) [reply f5_cccl/bigip.py]: OK. Let me do some research on this -- either it's a bug or there's something we don't quite understand.

Comment (edarzins) [reply f5_cccl/bigip.py]: Let's keep your workaround for now, but please add a comment that this is a workaround.

File: f5_cccl/schemas/cccl-net-api-schema.yml [hunk 2]
```diff
@@ -88,3 +88,8 @@ properties:
     items:
       $ref: "#/definitions/fdbTunnelType"
     type: "array"
+  userFdbTunnels:
+    description: "List of user-created FDB tunnel resources to be updated"
```
Comment (edarzins) [inline f5_cccl/schemas/cccl-net-api-schema.yml]: Expand the description for userFdbTunnels, i.e. that these are expected to be administratively-created beforehand, CCCL will perform updates only (no deletion or creation), etc

**Comment by sjberman** [comment]
Had to fix a few grammar/flake8 issues, but everything should be good to go now.

**Pull request merged**